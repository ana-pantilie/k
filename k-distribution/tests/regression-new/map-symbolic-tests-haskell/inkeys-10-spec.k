// Copyright (c) 2019 K Team. All Rights Reserved.
// generated by gen-tests.sh

requires "map-tests.k"

module INKEYS-10-SPEC
    imports MAP-TESTS

    claim <k> inkeys ( Y:MyId in_keys ( Z:MyId |-> 1 M:Map X:MyId |-> 0 ) ) => inkeysResult ( ?_ ) </k>
        requires Y in_keys(M)
    // TODO: result should be true

endmodule
