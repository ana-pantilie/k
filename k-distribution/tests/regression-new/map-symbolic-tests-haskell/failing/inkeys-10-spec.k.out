  #Not ( {
    X
  #Equals
    Z
  } )
#And
  #Not ( {
    true
  #Equals
    Y:MyId in_keys ( M
    X:MyId |-> 0
    ( Z:MyId |-> 1 ) )
  } )
#And
  <k>
    inkeysResult ( Y:MyId in_keys ( M
    X:MyId |-> 0
    ( Z:MyId |-> 1 ) ) ) ~> .
  </k>
#And
  {
    false
  #Equals
    X:MyId in_keys ( M )
  }
#And
  {
    false
  #Equals
    Z:MyId in_keys ( M )
  }
#And
  {
    true
  #Equals
    Y:MyId in_keys ( M )
  }
